util/dvsim/Deploy.py

```diff
@@ -240,16 +240,28 @@ def set_status(self):
         self.status = 'P'
         if self.dry_run is False:
             for fail_pattern in self.fail_patterns:
-                grep_cmd = "grep -m 1 -E \'" + fail_pattern + "\' " + self.log
-                (status, rslt) = subprocess.getstatusoutput(grep_cmd + " -c")
-                if rslt != "0":
-                    (status, rslt) = subprocess.getstatusoutput(grep_cmd)
+                # return error messege with the following 4 lines
+                grep_cmd = "grep -m 1 -A 4 -E \'" + fail_pattern + "\' " + self.log
+                (status, rslt) = subprocess.getstatusoutput(grep_cmd)
+                if rslt:
                     msg = "```\n{}\n```\n".format(rslt)
                     self.fail_msg += msg
                     log.log(VERBOSE, msg)
                     self.status = 'F'
                     break
 
+            # if returncode fails, usually encounter compile error, null pointer
```

Comment:
I think this sentence is not always true. Xcelium returns with non-zero exit code even if UVM_ERRORs are encountered.

Reply:
Perhaps just say, "If fail patterns were not encountered, but the job returned with non-zero exit code for whatever reason, then show the last 10 lines of the log as the failure message, which might help with the debug."

Reply:
Actually, xcelium returns zero exit code if it's just UVM_ERROR, which is the same as VCS. Most of i2c errors are due to null pointer error and sim exits immediately after the error.

I guess this kind of printing last 10 lines will be very rare as we list most of the error patterns. You comment makes sense, only very usual error will show last 10 lines of log.

Reply:
I see, thanks.